clowder-framework/clowder2
# Registration with keycloak #3

To test:
1. import new realm definition and double check if 1) registraion is enabled in keycloak and 2) the redirect URL is correct 
[realm-export-latest.json.zip](https://github.com/clowder-framework/clowder2/files/8663205/realm-export-latest.json.zip)
<img width="450" alt="image" src="https://user-images.githubusercontent.com/13950475/167674110-272fbefc-7dc5-451a-8380-8bd2d502e165.png">
<img width="1093" alt="image" src="https://user-images.githubusercontent.com/13950475/167674152-a703f60f-9a8e-4499-a76b-a123957b95ed.png">

2. Either go to http://localhost:3000/auth/logout and find "register" on the topbar; or directly go to http://localhost:3000/auth/register endpoint to register
<img width="1788" alt="image" src="https://user-images.githubusercontent.com/13950475/167673808-de6e8c27-b252-445f-b382-378df7dedc68.png">

4. You can also test registration by going to keycloak's login page and click "new user? register"
<img width="554" alt="image" src="https://user-images.githubusercontent.com/13950475/167673685-3806f7a6-32bf-44e8-802a-157242fd73b1.png">

 Note: I also fixed the "name" null null error on each dataset/file/folders
<img width="346" alt="image" src="https://user-images.githubusercontent.com/13950475/167673975-e72bf39f-0e31-462a-8af5-11650bcf1823.png">
